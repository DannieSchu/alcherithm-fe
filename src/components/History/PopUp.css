.PopUp {
  position: fixed;
  top: 10%;
  left: 25%;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: solid .1vw rgb(197, 197, 197);
  border-radius: 1rem;
  width: 50%;
  margin-bottom: 1rem;
  padding-top: 1rem;
  border: 1px solid var(--gray);
  border-radius: 5px;
  box-shadow: var(--box-shadow);  
  min-width: 35rem;
  padding: 1rem;
  background-color: rgba(48, 198, 232, 0.6);
  
  article {
    width: 505px;
    margin-left: 0;
    padding-left: 0;
  }

  span:first-child {
    position: absolute;
    right: 2rem;
    cursor: pointer;
  }

  article > div {
    margin: 2rem;
  }

  h3 {
    color: white;
    margin-left: 2rem;
  }
}

.PopUp:hover {
  border: 1px solid rgb(197, 197, 197);
  box-shadow: var(--hover-box-shadow);
}
