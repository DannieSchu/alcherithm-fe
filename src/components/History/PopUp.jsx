import React from 'react';
import PropTypes from 'prop-types';
import Editor from '../Editors/Editor';
import styles from './PopUp.css';

const PopUp = ({ toggle, solutions }) => {
  const solutionElements = solutions.map((solution, i) => (
    <div key={i} >
      <Editor code={solution} readOnly={true} />
    </div>
  ));

  return (
    <section className={styles.PopUp}>
      <span onClick={() => toggle()}>✖️</span>
      <article>
        <h3>Your Solutions</h3>
        {solutionElements};
      </article>
    </section>
  );
};

PopUp.propTypes = {
  solutions: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  toggle: PropTypes.func.isRequired
};

export default PopUp;
