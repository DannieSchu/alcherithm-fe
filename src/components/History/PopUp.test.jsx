import React from 'react';
import { shallow } from 'enzyme';
import PopUp from './PopUp';

describe('PopUp component', () => {
  it('renders PopUp', () => {  
    const selectedSolutions = [`
    const removeOne = (num, arr) => {
    if(num % 3 === 2) {
        arr.pop();  
    }
    };
    
    const removeElements = (arr, callback) => {
    for(let i = 0; i < arr.length; i++) {
        const arrayItem = arr[i];
        callback(arrayItem, arr);
    }
    return arr;
    };
    `, `
    const removeOne = (num, arr) => {
    if(num % 3 === 2) {
        arr.pop();  
    }
    };
    
    const removeElements = (arr, callback) => {
    for(let i = 0; i < arr.length; i++) {
        const arrayItem = arr[i];
        callback(arrayItem, arr);
    }
    return arr;
    };
    `];

    const wrapper = shallow(<PopUp solutions={selectedSolutions} toggle={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
