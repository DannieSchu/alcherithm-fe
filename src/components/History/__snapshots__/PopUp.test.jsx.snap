// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp component renders PopUp 1`] = `
<section
  className="PopUp"
>
  <span
    onClick={[Function]}
  >
    ✖️
  </span>
  <article>
    <h3>
      Your Solutions
    </h3>
    <div
      key="0"
    >
      <Editor
        code="
    const removeOne = (num, arr) => {
    if(num % 3 === 2) {
        arr.pop();  
    }
    };
    
    const removeElements = (arr, callback) => {
    for(let i = 0; i < arr.length; i++) {
        const arrayItem = arr[i];
        callback(arrayItem, arr);
    }
    return arr;
    };
    "
        readOnly={true}
      />
    </div>
    <div
      key="1"
    >
      <Editor
        code="
    const removeOne = (num, arr) => {
    if(num % 3 === 2) {
        arr.pop();  
    }
    };
    
    const removeElements = (arr, callback) => {
    for(let i = 0; i < arr.length; i++) {
        const arrayItem = arr[i];
        callback(arrayItem, arr);
    }
    return arr;
    };
    "
        readOnly={true}
      />
    </div>
    ;
  </article>
</section>
`;
